packages/core/core-flows/src/cart/workflows/add-shipping-method-to-cart.ts

```diff
@@ -80,14 +80,15 @@ export const addShippingMethodToCartWorkflow = createWorkflow(
             provider_id: shippingOption?.provider_id,
             option_data: shippingOption?.data ?? {},
             method_data: inputOption.data ?? {},
+            from_location: shippingOption?.stock_location ?? {},
```

Comment:
nit: I believe this should be part of the `context`, alongside the cart data to be consistent with how this is otherwise used

Reply:
so you would pass a Record<option_id, stockLocaition> in context since we can have different locations for each option?

Reply:
I think I would move the context to be a part of the array, instead of being a "global" context for all options:
```
export interface ValidateShippingMethodsDataInput {
  options_to_validate: {
    id: string
    provider_id: string
    option_data: Record<string, unknown>
    method_data: Record<string, unknown>
    context: Record<string, unknown>
  }[]
}

const validatedData = await promiseAll(
  options_to_validate.map(async (option) => {
    const validated = await fulfillmentModule.validateFulfillmentData(
      option.provider_id,
      option.option_data,
      option.method_data,
      option.context
    )

    return {
      [option.id]: validated,
    }
  })
)
```

What do you think?

Reply:
IIRC, we pass the `from_location` and `...cart` in the context to other methods of the fulfillment methods right?

Reply:
ahh the other way around, yea think this is cleaner, pushed a change 👍